[Tests] S3 test failures

### Description

Various S3 test failures

```------- Stdout: -------
=== RUN   TestAccS3BucketLogging_TargetGrantByID
=== PAUSE TestAccS3BucketLogging_TargetGrantByID
=== CONT  TestAccS3BucketLogging_TargetGrantByID
bucket_logging_test.go:132: Step 1/5 error: Error running apply: exit status 1
Error: error putting S3 bucket (tf-acc-test-8809545006522085038) logging: AccessControlListNotSupported: Target grants not allowed for bucket owner enforced buckets
  status code: 400, request id: EHES1JC433XM83NV, host id: omuwYCClFETDoS+2Fn2Mzp71F2fyt+WgxXmYK9R149wyxuRXRogHskNkOj0t4biahu5OClgeiw0=
with aws_s3_bucket_logging.test,
on terraform_plugin_test.tf line 33, in resource "aws_s3_bucket_logging" "test":
33: resource "aws_s3_bucket_logging" "test" {
--- FAIL: TestAccS3BucketLogging_TargetGrantByID (107.40s)
FAIL
```

### References

_No response_

### Would you like to implement a fix?

None

--- a/internal/service/s3/bucket_logging_test.go
+++ b/internal/service/s3/bucket_logging_test.go
@@ -556,7 +556,7 @@ resource "aws_s3_bucket" "test" {
 resource "aws_s3_bucket_ownership_controls" "test" {
   bucket = aws_s3_bucket.test.id
   rule {
-    object_ownership = "BucketOwnerPreferred"
+    object_ownership = "ObjectWriter"
   }
 }
 
@@ -654,7 +654,7 @@ resource "aws_s3_bucket" "test" {
 resource "aws_s3_bucket_ownership_controls" "test" {
   bucket = aws_s3_bucket.test.id
   rule {
-    object_ownership = "BucketOwnerPreferred"
+    object_ownership = "ObjectWriter"
   }
 }
 

--- a/internal/service/s3/bucket_test.go
+++ b/internal/service/s3/bucket_test.go
@@ -423,7 +423,7 @@ func TestAccS3Bucket_Duplicate_UsEast1AltAccount(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				Config:      testAccBucketConfig_duplicateAltAccount(endpoints.UsEast1RegionID, bucketName),
-				ExpectError: regexp.MustCompile(s3.ErrCodeBucketAlreadyOwnedByYou),
+				ExpectError: regexp.MustCompile(s3.ErrCodeBucketAlreadyExists),
 			},
 		},
 	})

